### DISCUSSION TITLE
Blank window instead of readings

### DISCUSSION THREAD

**Discussion by Inkalika** [j-kaltes/Juggluco#127] [General]
After Sensor Error 373 the sensor readings are not readable. After an hour the readings are received via NFC, but they are not displayed on the main window. After rebooting the phone, the error type and sensor number disappear. There is a blank screen at this point. But if I turn on another phone, then on the new phone readings start to be displayed and change. Helps a complete reinstallation, change of version. But this time it didn't help.
Program version 7.4.6, 7.3.1. Libre 2 sensor
This situation occurs already with the second sensor in a row

Is it acceptable to turn on two phones to poll one sensor via bluetooth? Can polling two phones via bluetooth of a sensor cause Sensor Errod (373) error?
Noticed that if 2 phones are enabled via bluetooth there is a Sensor Error, but the NFC readings are coming through on both phones.

![photo_2024-02-20_17-27-48](https://github.com/j-kaltes/Juggluco/assets/89869109/7ee972b8-58f2-41a5-a395-aa306dc2aabc)
![photo_2024-02-20_17-28-08](https://github.com/j-kaltes/Juggluco/assets/89869109/359610d6-71da-4fff-b545-c2071c5fc2f2)
![photo_2024-02-20_17-28-58](https://github.com/j-kaltes/Juggluco/assets/89869109/1cc5eb4c-396b-49fb-a987-56e1297e8920)

**Comment by j-kaltes** [comment]
A sensor can connect via Bluetooth only with one device at a time. See  https://www.juggluco.nl/Juggluco/index.html#multiple
You can scan via NFC with multiple devices. You can also use one device via Bluetooth and then another device as long as you turn off Juggluco on the first device or turn off Bluetooth or "Use Bluetooth" and before scanning the sensor on the new device terminate the sensor on Juggluco on that device so that the sensor is initialized a new.
On the last screenshot you have disable Bluetooth so that will never work.

**Reply by Inkalika** [reply]
Yes, I turned off the bluetooth before making the screenshot (I made it at 17.30). This situation occurs with the second sensor and even after installing version 5.1 I also have no information about the sensor - no serial number, no readings. At the same time after turning off the bluetooth on this phone and turning it on on Redmy 9C on the new phone the readings go on. And I don't know what to do except reset the phone to factory settings.

**Reply by j-kaltes** [reply]
I see no problem. It sounds like it behaves exactly the way it should behave.

**Reply by Inkalika** [reply]
I assume that once the error is gone, the readings will start transmitting to the phone again, but they don't come over bluetooth, but are only read over NFC. But if I turn off this phone and turn on another one, the new one receives information via bluetooth. Previously, reinstalling the program, updating the program version helped, so that the readings started coming to the old phone via bluetooth again. Now with bluetooth turned on the readings do not come.

**Reply by Inkalika** [reply]
And on the right side where the sensor readings are, I usually see readings, serial number, errors like No connection, Sensor Error, Disable bluetooth. And in my case, it's a black screen.
In what cases can there be a situation that there is no sensor number, although in the sensor information it is present and there is an error log with it?

**Reply by j-kaltes** [reply]
When you turn off "_Use Bluetooth_" no information is shown at the right of the display and in left menu->Sensor no sensor id is shown.

**Reply by Inkalika** [reply]
I would like to install a log version to record this situation, but I can't figure out how I can create a log of the program using the adb shell run-as command tk.glucodata cat /data/data/tk.glucodata/files/logs/trace.log > trace.log Where do I need to copy it?

**Reply by Inkalika** [reply]
That's weird. Because when the bluetooth is off, on the right side I see the phrase "Enable bluetooth" (or Desable bluetooth?) . And after turning bluetooth on, the screen stays in this place without any readings or serial number. But on the other phone the readings start coming right away.

**Reply by j-kaltes** [reply]
As said, you can't connect two phones to the same sensor.

After turning on USB debugging in developers options,  you connect the phone to your computer, open a terminal and confirm that the phone is connected by typing "adb devices". You go to a writable directory and press:
```bash
adb shell run-as tk.glucodata cat /data/data/tk.glucodata/files/logs/trace.log > trace.log
```
trace.log is than written in that directory.

**Reply by Inkalika** [reply]
Before turning on the bluetooth on the old phone, I turn off the bluetooth on the new phone, but even after 10 minutes after turning on the bluetooth, the readings are not coming. But if I turn off the bluetooth on the old phone and turn it on on the new phone, the readings come to the new phone very quickly.

**Reply by j-kaltes** [reply]
Yes, you should scan again to make it work and either wait a long time before scanning or terminate the sensor on that phone and scan again. Bluetooth and "Sensor via Bluetooth" should also be turned on. There are also phones with bad Bluetooth that only very reluctantly make a connection with the sensor, see https://github.com/j-kaltes/Juggluco/issues/76#issuecomment-1699818968
Also when switching to a phone with good Bluetooth, it can take a terrible long time before it is working again after having scanned the sensor with another phone. This is totally different from switching between a phone and a watch that are connected via a mirror connection, thus having the same authentication information.

**Reply by Inkalika** [reply]
Three phones:
1) Redmi 12C After restoring the NFC bluetooth readings transmission, it still did not start transmitting (waited for more than 10 minutes). Black screen was displayed on the right side of the program.
2) Redmy 9C This phone was read the sensor when the first phone did not transmit readings via bluetooth for a long time. The readings came quickly
3) Samsung A12 One bluetooth reading came in, then Sensor Error appeared for a few minutes. One reading came back and then the error appeared again.
Ended up having to opt for a second phone until a new sensor is installed
